slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Importer.java

```diff
@@ -100,22 +101,24 @@ private void importDataInternal(final InputStream input, final Optional<List<Str
       }
 
       final ArrayNode dataNode = rootNode.withArray("data");
-      jdbi.useTransaction(
-          h -> {
-            for (int i = 0; i < dataNode.size(); i++) {
-              try {
-                final JsonNode validatorNode = dataNode.get(i);
-                parseValidator(h, validatorNode, pubkeys);
-              } catch (final IllegalArgumentException e) {
-                LOG.error("Failed to parse validator {}, due to {}", i, e.getMessage());
-                throw e;
-              }
-            }
-          });
+      IntStream.range(0, dataNode.size())
+          .parallel()
+          .forEach(
+              i -> {
+                try {
+                  jdbi.useTransaction(
+                      h -> {
+                        final JsonNode validatorNode = dataNode.get(i);
+                        importValidator(h, validatorNode, pubkeys);
+                      });
+                } catch (Exception e) {
```

Comment:
nit: `final` in catch block. Also, I would append `e.getMessage()` with the error log to have a better idea what went wrong during the transaction.

Reply:
good idea. done